src/CSET/operators/mesoscale.py

```diff
@@ -0,0 +1,123 @@
+        Iris cube containing data to smooth, supporting multiple dimensions
+        (at least two spatial dimensions must be supplied, i.e. 2D).
+    Gaussian_filter: boolean
+        Switch to determine if a Gaussian filter is applied.
+        If set to True a Gaussian filter is applied; if set to False
+        a Uniform filter is applied.
+        Default is True.
+    filter_scale: int
+        Scale at which to define the filter in grid boxes. If the
+        filter is a Gaussian convolution this value represents the
+        halfwidth of the Gaussian kernel.
+        Default is 40 grid points.
+
+    Returns
+    -------
+    An iris cube of the spatial perturbation field.
+
+    Return type
+    -----------
+    iris.cube.Cube
+
+    Notes
+    -----
+    In mesoscale meteorology the perturbation field is more important than the
+    balanced flows for process understanding. This function is designed
+    to create spatial perturbation fields based on smoothing with a Gaussian
+    kernel or a uniform kernel.
+
+    The kernels are defined by the filter_scale, which for mesoscale
+    perturbations should be between an approximate cloud separation distance,
+    of 30 km, and synoptic scale variations (1000 km). In practice any
+    value between these ranges should provided broadly consistent results (e.g.
+    [Flacketal2016]_). The Gaussian kernel will give greater importance to
+    areas closer to the event and will produce a smooth perturbation field.
+    The uniform kernel will produce a smooth perturbation field but will not
+    give local features as much prominence.
+
+    Caution should be applied to boundaries, particularly if the domain is of
+    variable resolution, as some numerical artifacts could be introduced.
+
+    References
+    ----------
+    .. [Flacketal2016] Flack, D.L.A., Plant, R.S., Gray, S.L., Lean, H.W.,
+       Keil, C. and Craig, G.C. (2016) "Characterisation of Convective
+       Regimes over the British Isles." Quarterly Journal of the Royal
+       Meteorological Society, vol. 142, 1541-1553. doi:10.1002/qj.2758
+
+    Examples
+    --------
+    >>> Temperature_perturbation = meso.spatial_perturbation_fields(Temp,
+                                   Gaussian_filter=True,filter_scale=40)
+    >>> iplt.pcolormesh(Temperature_perturabtion[0,:,:],cmap=mpl.cm.bwr)
+    >>> plt.gca().coastlines('10m')
+    >>> plt.clim(-5,5)
+    >>> plt.colorbar()
+    >>> plt.show()
+
+    """
+    pert_field = original_field.copy()
+    # find axes of spatial coordinates in field
+    coords = [coord.name() for coord in original_field.coords()]
+    # axes tuple containing latitude, longitude coordinate name.
+    axes = (
+        coords.index(get_cube_yxcoordname(original_field)[0]),
+        coords.index(get_cube_yxcoordname(original_field)[1]),
+    )
+    # apply convolution depending on type used
+    if Gaussian_filter:
+        filter_type = "Gaussian"
+        logging.info("Gaussian filter applied.")
```

Comment:
```suggestion
        logging.debug("Gaussian filter applied.")
```

Reply:
Not debugging information, just telling the user what filter has been applied therefore info used.